tests/test_items.py

```diff
@@ -615,3 +615,20 @@ def test_metadata():
         obj2.metadata = Metadata(dateDownloaded="foo")
         assert type(obj2.metadata) == metadata_cls
         assert obj2.metadata.dateDownloaded == "foo"
+
+
+def test_request():
+    """Test request class conversion on assignment"""
+
+    product_navigation = ProductNavigation(**_PRODUCT_NAVIGATION_MIN_KWARGS)
+    product_navigation.subCategories = [
+        Request(  # type: ignore[list-item]
```

Comment:
If mypy doesn't like it here, it won't like it in the user code as well. Is there a way to make mypy happy?

Reply:
Tools don't know about `attrs` converters so this is expected, though I'm not sure why mypy doesn't consider other uses errors. In this case it's a more specific check about list items, not just e.g. `BasePage("http://example.com")`.